### PR TITLE
Weakpoint proficiencies: (some) zombies and insects

### PR DESCRIPTION
**Pull request by dseguin**
#### Summary
None

#### Purpose of change
Even though Joshua-Chin added the infrastructure in #52081, no weakpoint proficiencies have ever been added to the game.

#### Describe the solution
Add proficiencies for the currently defined weakpoints (which as of now only includes some zombies and some insects).

Weakpoint proficiencies are defined per monster using the `"families"` field. These proficiencies can be practiced by dissecting monsters that define that field:

dissect_msg
learned_profs

The bonus/penalty values are based on `melee_skill`, which is used along with proficiencies in `weakpoint_attack::compute_wp_skill` to determine weakpoint weapon skill:

https://github.com/CleverRaven/Cataclysm-DDA/blob/3d0002bc4491ffed11874cd2031c10478ead5891/src/mtype.h#L328

https://github.com/CleverRaven/Cataclysm-DDA/blob/3d0002bc4491ffed11874cd2031c10478ead5891/src/weakpoint.cpp#L72-L75

#### Describe alternatives you've considered
There's certainly more weakpoints and proficiencies that could be added, but I don't have an eye for balance :P

#### Testing
See screenshots above. Also tested by stepping through `weakpoint_attack::compute_wp_skill` and `monster::absorb_hit` to see how the modified wp skill affects the weakpoint selection.

#### Additional context
Here's a list of weakpoint proficiencies that were newly added:

| Proficiency ID | Proficiency name | Usage*
|--- |--- |---
| `prof_wp_zombie` | Principles of Zombie Biology | All zombies
| `prof_wp_syn_armored` | Synthetic Armors | Creatures wearing man-made armours
| `prof_wp_nat_armored` | Natural Armors | Creatures with biological armour
| `prof_wp_basic_bug` | Invertibrate Biology | Insects, arachnids, worms, etc.
| `prof_wp_flying` | Creatures of Flight | All flying creatures
| `prof_wp_skeleton` | Natural Exoskeletons | "Bony" creatures
| `prof_wp_child` | Small Humanoids | Child creatures (ie: zombie children)
| `prof_wp_hulk` | Large Humanoids | Hulks

Additionally, here are some existing proficiencies that have been applied to weakpoints:

| Proficiency ID | Proficiency name | Usage*
|--- |--- |---
| `prof_intro_biology` | Principles of Biology | Common requirement for biological weakpoints
| `prof_biochemistry` | Biochemistry | Creatures producing certain chemicals (ie: acid creatures)
| `prof_physiology` | Physiology | Non-zombie humans (ie: ferals)
| `prof_electromagnetics` | Electromagnetics | Electrical creatures
| `prof_burn_care` | Burn Care | Burned/charred creatures

*\* Applied to monsters with existing weakpoints only*

Hopefully this inspires someone more balance-oriented to add some more weakpoints and weakpoint proficiencies

### CODE REVIEW

**Comment by Shodan14** [comment]
We're going to need a better way to organize the proficiencies list with that..

**Comment by bananapheasant** [comment]
i fully second some new way to view and organize proficiencies as it becomes more comprehensive. Proficiencies having levels like skills would perhaps be a good direction to head as well. Currently many have beginner, advanced, etc. Just viewing the proficiencies and their level, possibly with a description of the advantages, would be cool.

**Comment by dseguin** [comment]
I hear you. I'm working on a UI to more clearly display proficiencies, which will be accessible from a shortcut in the @ menu. That will come in a separate PR.

**Comment by TheShadowFerret** [comment]
> This PR adds proficiencies about the PC knowing where to strike to hurt a particuliar creature the most. Those you quoted seems like more about knowing what to do with their remains

Dissection of remains also trains these, I'm not seeing a significant difference here.

**Comment by dseguin** [comment]
> We already have Principles of Biology, Physiology, and Xenology as proficiencies. Shouldn't they be used instead of using _only_ new proficiencies?

I'm not sure, to be honest. They didn't seem appropriate for zombies or insects, but Xenology could be appropriate for migo weakpoints for example.

Maybe I'm wrong, though. I can be convinced otherwise.

**Comment by dseguin** [comment]
I updated the weakpoint proficiencies to use more original names/descriptions and reused some existing proficiencies (list included in PR description). Erk suggested a distribution that makes much more sense than what I was using.

C.C.: @I-am-Erk

File: data/json/proficiencies/weakpoints.json [hunk 1]
```diff
@@ -19,5 +19,93 @@
...
+    "default_weakpoint_penalty": 0
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_wp_hulk",
+    "category": "prof_weakpoint",
+    "name": { "str": "Large Humanoids" },
+    "description": "You have some practice in fighting large hulking humanoids.  You're better at reaching certain weak points.",
+    "can_learn": true,
+    "time_to_learn": "6 h",
```
Comment (Termineitor244) [inline data/json/proficiencies/weakpoints.json:107]:
```suggestion
    "time_to_learn": "6 h",
    "required_proficiencies": [ "prof_intro_biology" ],
```

File: data/json/proficiencies/weakpoints.json [hunk 2]
```diff
@@ -19,5 +19,93 @@
...
+    "default_weakpoint_penalty": -2
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_wp_child",
+    "category": "prof_weakpoint",
+    "name": { "str": "Small Humanoids" },
+    "description": "You have some practice in fighting small humanoids like children.  You don't necessarily feel good about it.",
+    "can_learn": true,
+    "time_to_learn": "4 h",
```
Comment (Termineitor244) [inline data/json/proficiencies/weakpoints.json:96]:
```suggestion
    "time_to_learn": "4 h",
    "required_proficiencies": [ "prof_intro_biology" ],
```

File: data/json/proficiencies/weakpoints.json [hunk 3]
```diff
@@ -19,5 +19,93 @@
...
+    "default_weakpoint_penalty": -4
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_wp_basic_bug",
+    "category": "prof_weakpoint",
+    "name": { "str": "Invertibrate Biology" },
+    "description": "Basic familiarity with various types of insects, arachnids, and worms.",
+    "can_learn": true,
+    "time_to_learn": "8 h",
```
Comment (Termineitor244) [inline data/json/proficiencies/weakpoints.json:63]:
```suggestion
    "time_to_learn": "8 h",
    "required_proficiencies": [ "prof_intro_biology" ],
```

File: data/json/proficiencies/weakpoints.json [hunk 4]
```diff
@@ -19,5 +19,93 @@
...
     "required_proficiencies": [ "prof_debug_weakpoint" ]
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_wp_zombie",
+    "category": "prof_weakpoint",
+    "name": { "str": "Principles of Zombie Biology" },
+    "description": "Very basic familiarity with various types of undead creatures.",
+    "can_learn": true,
+    "time_to_learn": "6 h",
```
Comment (Termineitor244) [inline data/json/proficiencies/weakpoints.json:30]:
```suggestion
    "time_to_learn": "6 h",
    "required_proficiencies": [ "prof_intro_biology" ],
```

**Review by Termineitor244 — CHANGES_REQUESTED** [review]
Looks pretty good to me! But I think there could be a basic/general weakpoint proficiency for zombies and ferals in general, you could use the "Principles of Biology" proficiency for that, since its related, and already used as a prerequisite for the already existing proficiencies.

You could decrease the time it takes to learn basic biology, and in turn increase the time for the most specific proficiencies, since they are most specific (and difficult) disciplines into biology, different than "If you hit the neck, it will hit harder", or "The legs are essential for balance, hit that", hulk biology would be more like "The hulks, because of their overgrown size and mutations, are stronger and more resistant overall, but after carefully analysis, it has been revealed that they lack balance/resistance in... areas, so a good and measured strike there could bring them down"

I'm not sure how the proficiencies code works, but I think the family entries of each monster would need "prof_intro_biology" added too for one to learn it (and then learn the specifics too)...
But that is already a problem, since some of the proficiencies require it, but you cannot learn it by combat.

**Comment by dseguin** [comment]
> I'm not sure how the proficiencies code works, but I think the family entries of each monster would need "prof_intro_biology" added too for one to learn it (and then learn the specifics too)... But that is already a problem, since some of the proficiencies require it, but you cannot learn it by combat.
> 
> Edit: I suggested changes in the proficiencies that I think should have "prof_intro_biology" as a requisite at the very least, the armor and flying proficiencies could do well without it, but these ones I feel would need it.

Thanks! Adding `prof_intro_biology` to the "families" field should be enough to make it learnable. You can't practice a proficiency if you don't have the pre-requisites, but adding the pre-requisite makes it so that you can practice it first before learning others (in this case via dissection).

**Comment by Termineitor244** [comment]
> Thanks! Adding `prof_intro_biology` to the "families" field should be enough to make it learnable. You can't practice a proficiency if you don't have the pre-requisites, but adding the pre-requisite makes it so that you can practice it first before learning others (in this case via dissection).

Good to know! And thanks for the good work!

**Pull request merged**